pkg/morph/subscriber/subscriber.go

```diff
@@ -91,20 +93,28 @@ func (s *subscriber) Close() {
 	s.client.Close()
 }
 
-func (s *subscriber) BlockNotifications() (<-chan *block.Block, error) {
+func (s *subscriber) BlockNotifications(rcv chan<- *block.Block) error {
+	s.Lock()
+	s.blockChan = rcv
+	s.Unlock()
+
 	if err := s.client.SubscribeForNewBlocks(); err != nil {
-		return nil, fmt.Errorf("could not subscribe for new block events: %w", err)
+		return fmt.Errorf("could not subscribe for new block events: %w", err)
 	}
 
-	return s.blockChan, nil
+	return nil
 }
 
-func (s *subscriber) SubscribeForNotaryRequests(mainTXSigner util.Uint160) (<-chan *result.NotaryRequestEvent, error) {
+func (s *subscriber) SubscribeForNotaryRequests(rcv chan<- *result.NotaryRequestEvent, mainTXSigner util.Uint160) error {
```

Comment:
Previously we could subscribe multiple times, but now we have a single channel. IMO this is a time-bomb.
I am also bothered about possible race-conditions  when we switch to another endpoint.
Not something we would like to do in the support branch.

Specifically: can we _prove_ that we cannot block when we read the channel, switch to another endpoint and then send a notifications?